components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/UsersResource.java

```diff
@@ -0,0 +1,170 @@
+import static org.wso2.micro.integrator.management.apis.Constants.PATTERN;
+import static org.wso2.micro.integrator.management.apis.Constants.ROLE;
+import static org.wso2.micro.integrator.management.apis.Constants.STATUS;
+import static org.wso2.micro.integrator.management.apis.Constants.USER_ID;
+
+/**
+ * Resource for a retrieving and adding users.
+ * <p>
+ * Handles resources in the form "management/users"
+ */
+public class UsersResource extends UserResource {
+
+    private static final Log LOG = LogFactory.getLog(UsersResource.class);
+
+    public UsersResource() {
+
+        methods = new HashSet<>();
+        methods.add(Constants.HTTP_GET);
+        methods.add(Constants.HTTP_POST);
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext, org.apache.axis2.context.MessageContext axis2MessageContext,
+                          SynapseConfiguration synapseConfiguration) {
+        String httpMethod = axis2MessageContext.getProperty(Constants.HTTP_METHOD_PROPERTY).toString();
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Handling " + httpMethod + "request.");
+        }
+        JSONObject response;
+        try {
+            switch (httpMethod) {
+                case Constants.HTTP_GET: {
+                    response = handleGet(messageContext);
+                    break;
+                }
+                case Constants.HTTP_POST: {
+                    response = handlePost(axis2MessageContext);
+                    break;
+                }
+                default: {
+                    response = Utils.createJsonError("Unsupported HTTP method, " + httpMethod + ". Only GET and "
+                                                     + "DELETE,  methods are supported",
+                                                     axis2MessageContext, BAD_REQUEST);
+                    break;
+                }
+            }
+        } catch (UserStoreException e) {
+            response = Utils.createJsonError("Error initializing the user store. Please try again later", e,
+                                             axis2MessageContext, INTERNAL_SERVER_ERROR);
+        } catch (IOException e) {
+            response = Utils.createJsonError("Error processing the request", e, axis2MessageContext, BAD_REQUEST);
+        }
+        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        Utils.setJsonPayLoad(axis2MessageContext, response);
+        return true;
+    }
+
+    protected JSONObject handleGet(MessageContext messageContext) throws UserStoreException {
+        String searchPattern = Utils.getQueryParameter(messageContext, PATTERN);
+        if (Objects.isNull(searchPattern)) {
+            searchPattern = "*";
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Searching for users with the pattern: " + searchPattern);
+        }
+        List<String> patternUsersList = Arrays.asList(getUserStore().listUsers(searchPattern, -1));
```

Comment:
Shall we move this line on top of the above debug log so that we can combine both debugs and don't need to check two times that debug is enabled.

Reply:
I added the 2 logs before and after searching the users from the store, since something could go wrong there.